### DISCUSSION TITLE
Deepgram Tier Issue

### DISCUSSION THREAD

**Discussion by RSMNYS** [deepgram/community#256] [General help]
### Which Deepgram product are you using?

Deepgram SDKs

### Details

Even though we use the nova tier in the setup, in the usage logs we can see that base tier is used. The costs for such requests usually 0, and quality of the transcription is bad. When send multiple requests with the same call record we could receive different results with the good quality, and bad quality. How to make sure the nova tier is always in use?

### If you are making a request to the Deepgram API, what is the full Deepgram URL you are making a request to?

POST /v1/listen?multichannel=true&smart_format=true&utterances=true&model=nova&utt_split=1.5&detect_language=true

### If you are making a request to the Deepgram API and have a request ID, please paste it below:

bb7f4033-33f6-4c0c-8fa1-e15956ce977f

### If possible, please attach your code or paste it into the text box.

```js
  const timestamp = Date.now();
  const timestampedUrl = `${recordUrl}?timestamp=${timestamp}`;
  const transcriptionResponse = await deepgram.transcription.preRecorded(
    { url: timestampedUrl },
    {
      multichannel: true,
      smart_format: true,
      utterances: true,
      model: "nova",
      utt_split: 1.5,
      detect_language: true
    }
  );
```

**Comment by jjmaldonis** [comment]
Hey @RSMNYS, the request you are making with that code will use the nova tier/model.

I took your query parameters and ran them through the Node SDK to reproduce your API call. Please see the code below. I then checked the usage logs and I see that the Nova tier is used (screenshot below). What does your usage logs show?

![image](https://github.com/deepgram/community/assets/2326898/e12d9754-ffb4-446e-bae8-852ce105ed32)


```typescript
import Deepgram from '@deepgram/sdk';
import { ErrorResponse, PrerecordedTranscriptionResponse } from '@deepgram/sdk/dist/types';
import querystring from "querystring";


const DEEPGRAM_API_KEY = process.env.DEEPGRAM_API_KEY as string;
const deepgram = new Deepgram.Deepgram(DEEPGRAM_API_KEY)

const audioUrl = "https://res.cloudinary.com/deepgram/video/upload/v1663090404/dg-audio/AI_Show_afpqeh.m4a"
const options = {
  multichannel: true,
  smart_format: true,
  utterances: true,
  model: "nova",
  utt_split: 1.5,
  detect_language: true
};
console.log(querystring.stringify(options));
await deepgram.transcription.preRecorded(
  { url: audioUrl },
  options,
).then((result: PrerecordedTranscriptionResponse | ErrorResponse) => {
  console.log(result["results"]["channels"][0]["alternatives"][0]["transcript"]);
});
```

**Reply by RSMNYS** [reply]
![Screenshot 2023-07-05 at 19 20 54](https://github.com/deepgram/community/assets/17128789/196b729d-8e76-494e-9576-799f0a31578f)

Thank you for your response. I'm attaching the screenshot of our logs. So you can see, even we send nova, the base tier is listed. 

If you provide your email, we can send you the call recording link, it's one of some that has issues.  Thanks.

**Comment by cateas** [comment]
Hi @jjmaldonis ,

As @RSMNYS mentioned, we are actually getting different results. Some are base, some enhanced, some nova.

Please check below, same file, one time we got base:
![image](https://github.com/deepgram/community/assets/16335463/1c9bcdc2-7376-4839-a899-2e041263fc3d)

The next time we run the same request, we got enhanced:
![image](https://github.com/deepgram/community/assets/16335463/f8f92635-a736-4795-bf7f-2a70333f0064)

As I noticed this happens mostly for the files with 1 speaker only. Please note that we are receiving audio files with 1 speaker or 2 speakers. This is the reason why we use multichannel=true. We do not know if the file has 1 speaker or 2 speakers.

Thank you!

**Reply by jjmaldonis** [reply]
Hey @cateas and @RSMNYS, we're pretty sure this is happening because `detect_language` is set to `true` and `nova` currently supports English and (now!) Spanish. So if the detected language is neither English nor Spanish, then the request will use the next-most-appropriate model to perform the transcription. In this case, it looks like both `base` and `enhanced` are being identified as the most appropriate model in different circumstances.

Do you know what language(s) are being spoken in the audio? And are the responses coming back with different values of `detected_language`?

**Reply by RSMNYS** [reply]
@jjmaldonis I've shared the audio file that we are having issues with on you email. In our recordings only English and Spanish languages are used.

Thanks.

**Reply by cateas** [reply]
@jjmaldonis is it possible to define only English and Spanish instead of detect_language ? We are going to support only 2 languages, but I saw there is an option to set either one language or detect language. If detect language creates troubles, is it possible to define then the languages manually? 

Thank you!

**Reply by jjmaldonis** [reply]
@RSMNYS the audio file you sent is being identified as one of three languages: English, Ukrainian, or Portuguese, and clearly the language is English. I'll try to find out more about why the language detection is failing, but it may have to do with the long periods of silence in the audio. That said, this confirms the language detection is not always identifying English or Spanish, so that's why you're seeing a non-Nova model being used for transcription.

@cateas that currently isn't possible but it's a great idea. I'll talk to our team about this and see how feasible it is.

**Reply by jjmaldonis** [reply]
@cateas this is on our roadmap to implement!

**Reply by cateas** [reply]
@jjmaldonis do you know when we can expect it ? Also a question please, at this point for some call recordings we are getting base or enhanced models. Enhanced are not that bad, so we can go with them, but base returns really bad output. We have added some logic to retry the recording if at least one of the returned models is: 1ebf0ea2-45b9-4945-9615-cfd247467a45 (which I noticed is associated with base model). I guess there are more models for base, could you help us with some logic to determine if it is a base model ? 


Also I noticed that if the base model is returned first, then the price is $0 for the transcription (I guess because the output is totally off),

![image](https://github.com/deepgram/community/assets/16335463/269825a3-025a-4742-8834-099691d54610)


but if another model is first and then the base one, then it has a price, even thought the output is not that good.

![image](https://github.com/deepgram/community/assets/16335463/3a509361-0237-4e81-bcc3-278012f94762)

Thanks!

**Reply by cateas** [reply]
Also one more thing, I have tried the same file in the Deepgram Playground from UI and it returned nova mode.
Looks like it is working good from UI Playground, but not via API:

   "model_info": {
      "3312b52b-d08e-4072-bfff-124c410b770d": {
        "name": "general-nova",
        "version": "2023-03-13.31000",
        "arch": "nova"
      }
    }

![image](https://github.com/deepgram/community/assets/16335463/4865d865-fe7e-44a4-9510-0866548b1fa9)

Not sure if this may help to fix the issue, but wanted to share some additional feedback.

**Reply by jjmaldonis** [reply]
Hey @cateas, we now support the ability to restrict language detection to a subset of languages. Let me know if you have any questions. Updated docs here: https://developers.deepgram.com/docs/language-detection

**Comment by jjmaldonis** [comment]
@cateas responding to each of these below:

> do you know when we can expect it?

I don't have an ETA but it is more than a month out. The functionality is part of a larger release that is in the works.

> Also a question please, at this point for some call recordings we are getting base or enhanced models. Enhanced are not that bad, so we can go with them, but base returns really bad output.

If either enhanced or base are being used then it should only be because the detected language isn't English or Spanish. In that case, the language will be something incorrect, and I would expect the transcript to be poor for both the enhanced and base models because they are using the incorrect language.
Are you seeing that the transcript is being transcribed in the correct language when the enhanced model is used?

> I guess there are more models for base, could you help us with some logic to determine if it is a base model?

Absolutely. The model IDs you see are part of the API response, and we include additional information in the response that tells you the name of the models that were used. Below is a sample response with the `metadata` object. You'll notice the `model_info` field contains the model ID and the name of the model. You can check if the model name contains the string `"base"` to if a base model was used.

```json
{
  "metadata": {
    "transaction_key": "deprecated",
    "request_id": "3ae50cc2-01bb-463a-9a5c-0972934a8748",
    "sha256": "656306c63a422b39de6f781b1338af39c1d86e1de5145853e26e6b4ae5421fbd",
    "created": "2023-07-06T20:59:45.004Z",
    "duration": 4.9999375,
    "channels": 2,
    "models": [
      "42eeb5c9-6ce4-4dd9-b1a6-22cbde4d724b",
      "3312b52b-d08e-4072-bfff-124c410b770d"
    ],
    "model_info": {
      "42eeb5c9-6ce4-4dd9-b1a6-22cbde4d724b": {
        "name": "general-enhanced",
        "version": "1982-08-12.24327",
        "arch": "polaris"
      },
      "3312b52b-d08e-4072-bfff-124c410b770d": {
        "name": "general-nova",
        "version": "2023-03-13.31000",
        "arch": "nova"
      }
    }
  }
}
```

> Also I noticed that if the base model is returned first, then the price is $0 for the transcription (I guess because the output is totally off), but if another model is first and then the base one, then it has a price, even thought the output is not that good.

I'm not sure what's going on with the price. We're working on some updates to how prices are displayed, so I'm guessing this will be more in the near-future.

> Also one more thing, I have tried the same file in the Deepgram Playground from UI and it returned nova mode.

Okay, this is good to know, thanks for the additional info. When I was testing, I ran the file @RSMNYS sent me through our API 15+ times. The audio was in English, and the `detect_language` feature correctly identified the language as English in most of the 15+ tests. It's possible that the API just happened to identify it correctly when you ran it through the Playground, in which case Nova+English would be used as the model.

**Comment by cateas** [comment]
`Are you seeing that the transcript is being transcribed in the correct language when the enhanced model is used?`

No, but we are using confidence for filtering the bad transcript. We are filtering out everything < 0.75. But base model still returns a lot of bad data with higher confidence. This is why it does not work for us.


`'You can check if the model name contains the string "base" to if a base model was used.'`

Hmm, I am not sure "base" is right. For example this one which is associated with base has name 'general'
'1ebf0ea2-45b9-4945-9615-cfd247467a45': { name: 'general', version: '2022-03-14.0', arch: 'vega' }

Should we just search by 'general' then ? Or should we search by arch: "vega" ? 
Could you please confirm?

**Reply by jjmaldonis** [reply]
Hey @cateas my bad, "general" is the right thing to check for rather than "base". You can ignore "vega".

**Reply by cateas** [reply]
Got it. Thanks!

**Comment by cateas** [comment]
@jjmaldonis is it possible to revise the pricing for our account? This issue generates a much higher cost for the calls which use base and enhanced tier instead of nova.

Thank you!

**Reply by jjmaldonis** [reply]
I'm still working on this. It may be worth it for you to separately reach out to our sales team for a call.

**Comment by cateas** [comment]
@jjmaldonis coming with an update related to this issue.
For the cases when it is single sided recording, meaning only one speaker is recorded, if we do not send multichannel parameter we get nova model. We do not know if the file will be with one speaker or 2 speakers.

I assume that what happens is the following, we send multi-channel parameter, Deepgram gets a file with one speaker, it assumes that there should be 2 speakers and for the second speaker where it is silence it tries to get a transcription. Since we also send detect language (cause we don not know if it is English or Spanish) it tries to find another language to fill the transcription for the second channel. My assumption is that this causes wrong language and base / enhanced model as result.

I hope this additional information will help to narrow down the issue.

Thanks!

**Reply by jjmaldonis** [reply]
This is very helpful, thanks!

**Reply by cateas** [reply]
@jjmaldonis even thought we are sending the right parameter multi-channel true or false, there are still situations when it returns a wrong tier. 
I have shared via email some of examples. Mostly it happens for cases when we have the ring tone, but no one takes the call, but there are situations when we have English and Spanish together (I shared some examples).

Thanks!

**Comment by cateas** [comment]
Thank you for update!

Best regards,


Kate Barari

Project Manager



Email:       ***@***.******@***.***>

Skype:      catiusel

________________________________
[https://nextcloud.scopicsoftware.com/s/r7RentpeWEXWFHN/preview?size_id=4]
Website <https://scopicsoftware.com/>       •    Portfolio<https://scopicsoftware.com/portfolio/>
​
[https://nextcloud.scopicsoftware.com/s/dY8jemD3my5dTFR/preview]<https://www.linkedin.com/company/5182084/admin/>  [https://nextcloud.scopicsoftware.com/s/85qJLKbsbP4YAS6/preview] <https://www.facebook.com/scopicsoftware>   [https://nextcloud.scopicsoftware.com/s/yK6HN8CsXgG88wd/preview] <https://twitter.com/scopicsoftware>   [https://nextcloud.scopicsoftware.com/s/2Cc68frJsMrM2Hw/preview] <https://www.instagram.com/accounts/login/?next=/scopic_software/>
• Become a Referral Partner <https://scopicsoftware.com/about-us/referral-program/> •
________________________________
